### PR TITLE
Race condition in lazy sink on immediate failure 

### LINKED ISSUE #25410: Stream hang with concatted lazy failed source fed into lazy inited sink
The following is a very round about way of achieving not much, but it's the best minimizer that I've been able to come up with:

```scala
  case object MyException extends Exception
  Await.result(
    Source(List(1, 2, 3))
      .concat(Source.lazily(() => Source.failed(MyException)))
      .runWith(Sink.lazyInitAsync(() => Future.successful(Sink.seq[Int])))
      .flatMap {
        case Some(future) => future
        case None => Future.successful(Seq())
      }.recover {
        case MyException => Seq()
      },
    1.second
  )
```

I expect the above program to finish with no error due to the `Future.recover` from `MyException`. However, it fails with a `TimeoutException` in Akka 2.5.14 on Scala 2.12.

### PR DESCRIPTION
**Pull request by johanandren**
Fix feels a bit like just doing something about the symptoms and not the root cause.

Fixes #25410

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/Sinks.scala [hunk 1]
```diff
@@ -544,7 +544,13 @@ import scala.util.{ Failure, Success, Try }
...
             // propagate exception irrespective if the cached element has been pushed or not
             subOutlet.fail(ex)
-            maybeCompleteStage()
+            // #25410 if we fail the stage here directly, the SubSource may not have been started yet,
+            // which can happen if upstream fails immediately after emitting a first value.
+            // The SubSource won't be started until the stream shuts down, which means downstream won't see the failure,
+            // scheduling it lets the interpreter first start the substream
+            getAsyncCallback[Throwable] {
+              case ex: Throwable ⇒ failStage(ex)
+            }.invoke(ex)
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/Sinks.scala]: `getAsyncCallback[Throwable](failStage).invoke(ex)` ?

**Review by patriknw — COMMENTED** [review]
an improvement over current situation I guess, might be good enough?

**Review by chbatey — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM (somewhat scary)

**Comment by jroper** [comment]
Thanks :)

**Pull request merged**